spec/parseResponse.spec.js

```diff
@@ -99,392 +101,55 @@ describe('parseResponse', function () {
-                    expectedAdEntry[i] = getExpectedAdEntry(mockData[i]);
-
-                    partnerModule.parseResponse(1, mockData[i], [returnParcels[i]]);
-
-                    for (var j = 0; j < expectedAdEntry[i].length; j++) {
-                        expect(registerAd).to.have.been.calledWith(sinon.match(expectedAdEntry[i][j]));
-                    }
-                }
-            }
-        });
+		
+		for (var i = 0; i < mockData.length; i++) {
+			(function(counter) {
+				var expectedResults = responseData.results[counter];
+				var requestId = expectedResults.requestId;
+				it("Check response for slot " + requestId, function() {
+					partnerModule.parseResponse(1, mockData[counter], returnParcels);
+					if (expectedResults.pass === false) {
+						chai.expect(registerAd.callCount).to.equal(1);
+						var registerAdArgs = registerAd.getCalls()[0].args;
+						chai.expect(registerAdArgs.price).to.equal(mockData.price);
+						chai.expect(registerAdArgs.dealId).to.equal(mockData.placementId);
+					}
+					for (var responseAttr in expectedResults) {
+						if (responseAttr === "targeting") {
+							chai.expect(expectedResults[responseAttr].ix_imdi_dealid, "(Request:" + requestId + ") Attribute " + responseAttr + ".ix_imdi_dealid should be " + JSON.stringify(returnParcels[counter][responseAttr].ix_imdi_dealid) + ".  Instead it is " + JSON.stringify(expectedResults[responseAttr].ix_imdi_dealid)).to.deep.equal(returnParcels[counter][responseAttr].ix_imdi_dealid);
+							chai.expect(expectedResults[responseAttr].ix_imdi_cpm, "(Request:" + requestId + ") Attribute " + responseAttr + ".ix_imdi_cpm should be " + JSON.stringify(returnParcels[counter][responseAttr].ix_imdi_cpm) + ".  Instead it is " + expectedResults[responseAttr].ix_imdi_cpm).to.deep.equal(returnParcels[counter][responseAttr].ix_imdi_cpm);
+							chai.expect(expectedResults[responseAttr].ix_imdi_id, "(Request:" + requestId + ") Attribute " + responseAttr + ".ix_imdi_id should be " + JSON.stringify(returnParcels[counter][responseAttr].ix_imdi_id) + ".  Instead it is " + expectedResults[responseAttr].ix_imdi_id).to.deep.equal(returnParcels[counter][responseAttr].ix_imdi_id);
+							chai.expect(expectedResults[responseAttr].pubKitAdId, "(Request:" + requestId + ") Attribute " + responseAttr + ".pubKitAdId should be a string starting with an underscore.  Instead it is " + expectedResults[responseAttr].pubKitAdId).to.match(/^_[a-zA-Z0-9]+$/);
+						} else {
+							chai.expect(returnParcels[counter][responseAttr], "(Request:" + requestId + ") Required attribute " + responseAttr + " does not exist").to.exist;
+							chai.expect(expectedResults[responseAttr], "(Request:" + requestId + ") Attribute " + responseAttr + " should be " + JSON.stringify(expectedResults[responseAttr]) + ".  Instead it is " + responseData.results[counter][responseAttr]).to.deep.equal(returnParcels[counter][responseAttr]);
+						}
+					}
+				});
+			})(i);
+		};
+		
+		after(function () {
```

Comment:
The unit test failed as the function after is undefined. Please ensure all unit tests pass.

Reply:
All unit tests do pass.  Running the following:

npm install
mocha spec/*

gives the following output:

  generateRequestObj
    √ MRA - should return a correctly formatted object
    √ should correctly build a url
    √ MRA - should return a correctly formatted object
    √ should correctly build a url
    √ MRA - should return a correctly formatted object
    √ should correctly build a url
    √ MRA - should return a correctly formatted object
    √ should correctly build a url
    √ MRA - should return a correctly formatted object
    √ should correctly build a url
    √ MRA - should return a correctly formatted object
    √ should correctly build a url
    √ MRA - should return a correctly formatted object
    √ should correctly build a url
    √ MRA - should return a correctly formatted object
    √ should correctly build a url
    √ MRA - should return a correctly formatted object
    √ should correctly build a url
    √ MRA - should return a correctly formatted object
    √ should correctly build a url
    √ MRA - should return a correctly formatted object
    √ should correctly build a url
    √ MRA - should return a correctly formatted object
    √ should correctly build a url

  parseResponse
    should correctly parse bids:
      √ Check response for slot _2222222222
      √ Check response for slot _2222222223
      √ Check response for slot _2222222224
      √ Check response for slot _2222222225
      √ Check response for slot _2222222226

  Partner Profile
    √ should be configured correctly


  30 passing (155ms)

Reply:
Looks like calling mocha directly worked but calling npm test didn't.  Latest submission fixes this.